logger.js

```diff
@@ -50,28 +52,39 @@ function Logger(opts) {
 
     this.statsd = opts.statsd;
 
-    var levels = extend(defaultLevels, opts.levels || {});
-    var path = opts.path || "";
+    var levels = this.levels = extend(defaultLevels, opts.levels || {});
+    this.path = opts.path = "";
 
-    Object.keys(levels)
-        .forEach(function (levelName) {
+    this.streamsByLevel = Object.keys(levels)
```

Comment:
can we do two loops. One to build the streamsByLevel cache and one to build the levels. Far more readable.